From 1464e4ae4a3a4c49c235e6cda77adb304d2db810 Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@foss.st.com>
Date: Thu, 12 Dec 2024 14:38:43 +0100
Subject: [PATCH] media: verisilicon: h1: add support of transform 8x8

Add support of transform 8x8 to encoder.

Change-Id: Id9036b622a29c8c06c6bf8c0596fd20d0e4c049b
Signed-off-by: Hugues Fruchet <hugues.fruchet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/425277
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Hugues FRUCHET <hugues.fruchet@st.com>
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
---
 drivers/media/platform/verisilicon/hantro_h1_h264_enc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/platform/verisilicon/hantro_h1_h264_enc.c b/drivers/media/platform/verisilicon/hantro_h1_h264_enc.c
index 51fddf03862a..4122084e4f13 100644
--- a/drivers/media/platform/verisilicon/hantro_h1_h264_enc.c
+++ b/drivers/media/platform/verisilicon/hantro_h1_h264_enc.c
@@ -143,6 +143,8 @@ static void hantro_h1_h264_enc_set_params(struct hantro_dev *vpu, struct hantro_
 		| H1_REG_ENC_CTRL2_INTRA16X16_MODE(h264_intra16_favor[qp]);
 	if (params->flags & V4L2_H264_ENCODE_FLAG_ENTROPY_CABAC)
 		reg |= H1_REG_ENC_CTRL2_ENTROPY_CABAC;
+	if (params->flags & V4L2_H264_ENCODE_FLAG_TRANSFORM_8X8_MODE)
+		reg |= H1_REG_ENC_CTRL2_TRANS8X8_MODE_EN;
 	vepu_write_relaxed(vpu, reg, H1_REG_ENC_CTRL2);
 
 	reg = H1_REG_ENC_CTRL3_MV_PENALTY_1P(h264_diff_mv_penalty[qp])
-- 
2.50.1

